oauth2_http/java/com/google/auth/oauth2/MetricsUtils.java

```diff
@@ -67,4 +68,44 @@ private static String getAuthLibraryVersion() {
     }
     return version;
   }
+
+  public enum RequestType {
+    ACCESS_TOKEN_REQUEST("at"),
+    ID_TOKEN_REQUEST("it"),
+    METADATA_SERVER_PIN("mds"),
+    UNSPECIFIED("unspecified");
+
+    private String label;
+
+    private RequestType(String label) {
+      this.label = label;
+    }
+
+    public String getLabel() {
+      return label;
+    }
+  }
+
+  static String getGoogleCredentialsMetricsHeader(
+      RequestType requestType, CredentialType credentialType) {
+    if (requestType == RequestType.UNSPECIFIED) {
+      return String.format(
+          "%s %s/%s",
+          MetricsUtils.getLanguageAndAuthLibraryVersions(), "cred-type", credentialType.getLabel());
+    }
+    if (credentialType == CredentialType.UNKNOWN) {
+      return String.format(
+          "%s %s/%s",
+          MetricsUtils.getLanguageAndAuthLibraryVersions(),
+          "auth-request-type",
+          requestType.getLabel());
+    }
+    return String.format(
+        "%s %s/%s %s/%s",
+        MetricsUtils.getLanguageAndAuthLibraryVersions(),
+        "auth-request-type",
```

Comment:
nit: Maybe `auth-request-type` and `cred-type` as package-private constants? It looks mostly to be contained, but may be useful in tests.